pkg/util/http_util.go

```diff
@@ -40,7 +40,12 @@ func DetectHTTPServerAvailability(url string, waitIntervalInSecond int, shouldAv
 	for {
 		<-ticker.C
 
-		_, err := cli.Get(url)
+		resp, err := cli.Get(url)
+		if resp != nil && resp.Body != nil {
+			if err := resp.Body.Close(); err != nil {
```

Comment:
nit: Should we do this just when err is nil?

Reply:
I think the caller is responsible for closing it as long as `resp.Body` is not nil. And typically this check implicitly means `err == nil`.